crates/astria-conductor/src/conductor/inner.rs

```diff
@@ -244,13 +252,57 @@ impl Conductor {
             info!("all tasks shut down regularly");
         }
         info!("shutting down");
+
+        finished_tasks.reverse();
+        // If any tasks failed and don't warrant a restart, return their error
+        for task_res in finished_tasks {
+            if let Err(err) = task_res {
+                let Some(task_name) = task_failed(&err) else {
+                    continue;
+                };
+                if task_name == "executor" && check_for_restart(&err) {
+                    return Ok(RestartOrShutdown::Restart);
+                }
+                return Err(err);
```

Comment:
As mentioned above, remove all of this. Especially avoid the parsing the error message. in `task_failed`